kubeconfig.go

```diff
@@ -0,0 +1,69 @@
+package kubeconfig
+
+import (
+	"context"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func NewG8sClientFromSecret(ctx context.Context, secretName, secretNamespace string) (versioned.Interface, error){
+	restConfig, err := getRESTConfigFromSecret(secretName, secretNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := versioned.NewForConfig(restConfig)
+	if err != nil {
+		return nil, microerror.Maskf(err, "rest.RESTClientFor")
+	}
+	return client, nil
+}
+
+func NewK8sClientFromSecret(ctx context.Context, secretName, secretNamespace string) (kubernetes.Interface, error) {
+	restConfig, err := getRESTConfigFromSecret(secretName, secretNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, microerror.Maskf(err, "rest.RESTClientFor")
+	}
+	return client, nil
+}
+
+func getRESTConfigFromSecret(secretName, secretNamespace string) (*rest.Config, error) {
```

Comment:
These should also be methods and the first argument should be the context.

Reply:
Where we are gonna use this context? even in  tenant cluster, I can't find any usage of it, https://github.com/giantswarm/tenantcluster/blob/master/tenantcluster.go
So I'm not getting why I should even pass down to the private method.

Reply:
I get by this blog, 🤦‍♂️ sorry. https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39

Reply:
> https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39

❤️ Reading up around Go and Kubernetes is always good.